src/components/element-editor/validation/validation.tsx

```diff
@@ -0,0 +1,105 @@
+    const [childEntryValues, setChildEntryValues] = useState<IValidation[]>();
+
+    // =========================================================================
+    // HELPER FUNCTIONS
+    // =========================================================================
+    function getOptionsByType(elementType: EElementType) {
+        switch (elementType) {
+            case EElementType.EMAIL:
+                return ELEMENT_VALIDATION_TYPES["Text field"][
+                    EElementType.EMAIL
+                ].validationTypes;
+            default:
+                return ["Select", "Type 1"];
+        }
+    }
+
+    function getMaxEntries(elementType: EElementType) {
+        switch (elementType) {
+            case EElementType.EMAIL:
+                return ELEMENT_VALIDATION_TYPES["Text field"][
+                    EElementType.EMAIL
+                ].maxEntries;
+            default:
+                return 6; // this is a arbitary value will be changed later on
+        }
+    }
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleChildChange = (index: number, newValue: IValidation) => {
+        setChildEntryValues((prevValues) => {
+            const updatedValues = [...prevValues];
+            updatedValues[index] = newValue;
+            return updatedValues;
+        });
+    };
+
+    const handleAddButtonClick = () => {
+        setChildEntryValues((prevValues) => [
+            ...prevValues,
+            {
+                validationType: "",
+                validationRule: "",
+                validationErrorMessage: "",
+            },
+        ]);
+    };
+
+    const handleDelete = (index: number) => {
+        setChildEntryValues((prevValues) =>
+            prevValues.filter((_, i) => i !== index)
+        );
+    };
+
+    // =========================================================================
+    // USE EFFECTS
+    // =========================================================================
+
+    useEffect(() => {
+        const element = focusedElement.element;
+        setChildEntryValues(element.validation);
+    }, [focusedElement.element]);
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+
+    const renderChildren = useCallback(() => {
+        return childEntryValues?.map((child, index) => (
+            <ValidationChild
+                key={`validation-entry-${index}`}
+                onDelete={() => handleDelete(index)}
+                onChange={(newValue) => handleChildChange(index, newValue)}
+                options={getOptionsByType(element.type)}
+                value={child}
+            />
+        ));
+    }, [childEntryValues, handleDelete, handleChildChange, element.type]);
```

Comment:
actually there is no need for `useCallback` because that merely memo-ise the function; it will still run on every render